test/registry/registryValidation.test.ts

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (c) 2021, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import { expect } from 'chai';
+import { MetadataRegistry } from '../../src';
+import { registry as defaultRegistry } from '../../src/registry/registry';
+import got from 'got';
+
+const apiVersionMax = 52;
+interface CoverageObject {
+  types: {
+    [key: string]: {
+      scratchDefinitions: {
+        professional: string;
+        group: string;
+        enterprise: string;
+        developer: string;
+      };
+      channels: {
+        metadataApi: boolean;
+        sourceTracking: boolean;
+        toolingApi: boolean;
+      };
+    };
+  };
+  versions: {
+    selected: number;
+    max: number;
+    min: number;
+  };
+}
+
+describe('Registry Validation', () => {
+  const registry = defaultRegistry as MetadataRegistry;
+  const typesWithChildren = Object.values(registry.types).filter((type) => type.children);
+
+  describe('every type from metadata coverage is in the SDR registry', () => {
+    before(async function () {
+      this.timeout(10000);
+      const metadataCoverage = JSON.parse(
+        (
+          await got(
+            `https://mdcoverage.secure.force.com/services/apexrest/report?version=${apiVersionMax}`
+          )
+        ).body
+      ) as CoverageObject;
+
+      // these include child types
+      const metadataApiTypes = Object.entries(metadataCoverage.types).filter(
+        ([key, value]) => value.channels.metadataApi && !key.endsWith('Settings')
+      );
+      expect(metadataApiTypes.length).to.be.greaterThan(200);
+
+      describe('generated type specs', () => {
```

Comment:
What if we generated 2 arrays of type names and just did a deep compare?

Reply:
I tried that but order matters on a deep compare.

Reply:
`.to.have.deep.members()` shouldn't care about order...or just sort the arrays.  It's a nit.  Just don't like the dummy test.

Reply:
oh, I never tried that before.  nice.